plopfile.js

```diff
@@ -0,0 +1,319 @@
+      message: 'Describe what the component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }],
+    actions: data => {
+      const actions = [{
+        type: 'add',
+        path: 'app/components/{{properCase name}}/index.js',
+        templateFile: data.type === 'ES6 Class' ? './webpack/templates/component/es6.js.hbs' : './webpack/templates/component/stateless.js.hbs',
+        abortOnFail: true
+      }, {
+        type: 'add',
+        path: 'app/components/{{properCase name}}/{{properCase name}}.test.js',
+        templateFile: './webpack/templates/component/component.test.js.hbs',
+        abortOnFail: true
+      }];
+
+      if (data.wantCSS) {
+        actions.push({
+          type: 'add',
+          path: 'app/components/{{properCase name}}/styles.css',
+          templateFile: './webpack/templates/component/styles.css.hbs',
+          abortOnFail: true
+        });
+      }
+
+      return actions;
+    }
+  });
+
+  // Container Component Generator
+  plop.setGenerator('container', {
+    description: 'Generate a Container Component',
+    prompts: [{
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'name is required';
+      }
+    }, {
+      type: 'input',
+      name: 'description',
+      message: 'Describe what the container component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }, {
+      type: 'confirm',
+      name: 'wantActionsAndReducer',
+      message: 'Do you want to generate corresponding actions/reducer files for the container?'
+    }, {
+      type: 'confirm',
+      name: 'wantSelector',
+      message: 'Do you want to add selectors to the connect method?'
+    }, {
+      type: 'list',
+      name: 'selectorType',
+      message: 'Select one option',
+      choices: [
+        { name: 'Select from already available selectors', value: 'old' },
+        { name: 'Generate new one', value: 'new' }
+      ],
+      when: answers => answers.wantSelector
+    }, {
+      type: 'checkbox',
+      name: 'selectors',
+      message: 'Choose the selectors which has to be added (select/deselect selectors by pressing space)',
+      choices: fs.readdirSync('app/selectors').map(dir => ({ name: dir.slice(0, -3), value: dir.slice(0, -3) })),
+      validate: value => {
+        if (value.length > 0) {
+          return true;
+        }
+        return 'atleast one selector must be selector';
```

Comment:
Typos, this should read: "At least one selector must be selected"

Maybe add a notice that one selects a selector with space as well?